src/modules/bspwm.cpp

```diff
@@ -236,6 +265,10 @@ namespace modules {
       if (tag[0] == 'm' || tag[0] == 'M') {
         m_monitors.emplace_back(factory_util::unique<bspwm_monitor>());
         m_monitors.back()->name = value;
+        m_monitors.back()->index = index;
+
+        if (not m_pinworkspaces)
+          index++;
```

Comment:
```suggestion
        index++;
```

With pin-workspaces, this loop will have only a single iteration anyway